Scripts/Restriction/ranacompiler.py

```diff
@@ -94,6 +94,8 @@ def read_enzyme_record(handle):
     for line in handle:
         key, value = line[:2], line[5:].rstrip()
         if key == "ID":
+            # Sync the replacement of "- and ." by "_" or enzyme will not be found in the bairoch list
+            value = value.replace("-", "_").replace(".", "_")
```

Comment:
I'm guessing you changed this as per @MarkusPiotrowski's comment https://github.com/biopython/biopython/issues/4437#issuecomment-1711617443 

However I'm concerned that in addition to fixing looking up the enzyme ID from the Bairoch file, it prevents access to the unmodified ID (e.g. in end user scripts).

This needs a test please.

Reply:
Never mind, in refreshing the REBASE files on https://github.com/biopython/biopython/pull/4687 I see we don't currently expose the original ID string anyway (just the numerical ID).

e.g. ``CviKI-1`` is only visible as ``CviKI_1``.

Reply:
Hi Peter,  Yes good catch from @MarkusPiotrowski back in September! 

It does not change (much) the output of the script, the enzyme is always added to the BioPython dictionary but if the enzyme exists in the Bairoch file, it can't look up the correct REBASE ID to add the proper URL in the Restriction_Dictionary.py. As in my comment below, I also added the proper change on the logging message.
BTW there are enzymes that might still get caught on this because they might not be in the bairoch file .. 

Agreed on doing more tests, specially that just today the new REBASE release is out, with several new changes.

By  "it prevents access to the unmodified ID (e.g. in end user scripts)." do you mean access from Restriction_Dictionary.py ? Because if the enzyme had a dash or a colon on the name, it was already replaced by the old script, this change only adds the proper ID (REBASE number) and allows creating the URL.